### PR TITLE
Bootstrap 4 upgrade & cleanup

### PR DESCRIPTION
**Pull request by jpeddicord**
Lots here. UI appears to function fine; some pieces are still ugly and will need more cleanup over time.

Dropped most in-lined assets: webpack can source those from npm modules.

Shuffled webpack's configuration to output resources in /res, since it's not just javascript in there anymore.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: browser/components/CLAEditor.tsx [hunk 1]
```diff
@@ -190,67 +189,65 @@ export default class CLAEditor extends React.Component<Props, State> {
...
+            <label>Point of Contact</label>
+            <br/>
+            <input type="text"  list="contact" className="form-control" id="contactName" name="contactName" placeholder={this.state.data.contact_name}/>
+            <datalist id="contact">
+              {this.state.data.display.poc.map(user => {
+                return (<option key={user} value={user}/>);
+              })}
+            </datalist>
+            <br/>
+            <label>Date Signed</label>
```
Comment (jamesiri) [inline browser/components/CLAEditor.tsx:229]: Your date changes cause the following error to be thrown on submit with empty date:
error:  error: invalid input syntax for type date: ""
    at Connection.parseE (/home/local/ANT/jamesiri/code/oss-contribution-tracker/node_modules/pg/lib/connection.js:539:11)
    at Connection.parseMessage (/home/local/ANT/jamesiri/code/oss-contribution-tracker/node_modules/pg/lib/connection.js:366:17)
    at Socket.<anonymous> (/home/local/ANT/jamesiri/code/oss-contribution-tracker/node_modules/pg/lib/connection.js:105:22)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at TCP.onread (net.js:594:20

File: browser/components/ContributionsTable.tsx [hunk 2]
```diff
@@ -125,7 +126,7 @@ class ContributionsTable extends React.Component<Props, State> {
 
   static renderTables = (name, item, tableProps) => {
     return (
-      <div className="container" key={Math.random()}>
+      <div key={name}>
         <h4>{name}</h4>
         <Table
```
Comment (jamesiri) [inline browser/components/ContributionsTable.tsx]: CSS changes causes the table header to cut off the bottom of the search input.

**Review by jamesiri — COMMENTED** [review]
Overall this is looking good and taking care of a few things that I was planning on doing later anyway. Just a couple of minor issues.

File: browser/components/EditContributionTable.tsx [hunk 3]
```diff
@@ -252,8 +252,10 @@ export default class EditContributionTable extends React.Component<Props, State>
             width={125}
           />
         </Table>
```
Comment (jamesiri) [inline browser/components/EditContributionTable.tsx:256]: CSS changes causes the table header to cut off the bottom of the search input.

File: browser/components/ContributionsTable.tsx [hunk 4]
```diff
@@ -206,44 +207,39 @@ class ContributionsTable extends React.Component<Props, State> {
...
-        <div className="col-md-2">
-        <br/>
-          <input type="text" id="projectLISearch" onInput={this.storeSearch} placeholder="Search projects.."/>
-          <ul id="projectUL" className="media-list">
-            {newList.map(name => (<li className="media" key={count++}><a href="#" onClick={() => this.updateSelectedProject(name)}>{name}</a></li>))}
-          </ul>
+          <div className="col-sm-3">
+            <input type="text" id="projectLISearch" className="form-control"
+              onInput={this.storeSearch} placeholder="Search projects" />
+            <div id="projectUL" className="list-group mt-1">
```
Comment (jamesiri) [inline browser/components/ContributionsTable.tsx]: This had a restricted height so you could scroll the list without moving the page. Will need that back before merge.

File: browser/components/EditContributionTable.tsx (hunk 3 above)
Comment (jpeddicord) [reply browser/components/EditContributionTable.tsx:256]: Hmm, not seeing that here. You'll have to show me that on Monday.

File: browser/components/ContributionsTable.tsx (hunk 4 above)
Comment (jpeddicord) [reply browser/components/ContributionsTable.tsx]: Ah, I did forget that. Adding.

File: browser/components/CLAEditor.tsx (hunk 1 above)
Comment (jpeddicord) [reply browser/components/CLAEditor.tsx:229]: Marking those as required.

**Pull request merged**